bundle/src/main/java/com/adobe/acs/commons/email/process/impl/SendTemplatedEmailProcess.java

```diff
@@ -274,6 +274,10 @@ public final void execute(WorkItem workItem, WorkflowSession workflowSession, Me
                     .getTitle());
             // Set workflow initiator
             wfParams.put(SendTemplatedEmailConstants.WF_INITIATOR, workItem.getWorkflow().getInitiator());
+            
+            if(workItem.getMetaDataMap().containsKey("comment")) {
+                wfParams.put(SendTemplatedEmailConstants.WF_STEP_COMMENT, workItem.getMetaDataMap().get("comment").toString());
```

Comment:
@drewrobinson I'm curious if maybe it makes sense to just shove the entire workItem.getMetadataMap() into the `wfParams` map. Similar to how the payload properties are converted to a `<String, String>` i.e 

```java
Map<String, String> workItemMetadata = SendTemplatedEmailUtils.getWorkItemMetadata(workItem, sdf);
            if (workItemMetadata != null) {
                emailParams.putAll(workItemMetadata);
            }
```
I think you may need to prefix the key with something like `workItem-` to ensure that you don't accidentally override a payload property key... but then again maybe this is overkill...

https://github.com/Adobe-Consulting-Services/acs-aem-commons/blob/master/bundle/src/main/java/com/adobe/acs/commons/email/process/impl/SendTemplatedEmailUtils.java#L69

Reply:
Great thought, @godanny86. I like that idea a lot and appreciate the feedback. I'll get started on the improvements.

Reply:
@drewrobinson Are you planning on coming up w/ a way to move all metadata properties over?

Also, IIRC,`putAll(..)` wont work due to the values of the metadata map not being actual Strings, but rather those weird Value proxy objects (i think we ran into this when you initially tried the templated email stuff?)

Reply:
its true, you would need to introduce an additional method to convert the metadataMap to a simple HashMap <String,String>. However you should be able to re-use some of the methods in  https://github.com/Adobe-Consulting-Services/acs-aem-commons/blob/master/bundle/src/main/java/com/adobe/acs/commons/email/process/impl/SendTemplatedEmailUtils.java to make it easier